vm/vm.go

```diff
@@ -14,20 +15,27 @@ import (
 )
 
 var MemoryBudget int = 1e6
-var errorType = reflect.TypeOf((*error)(nil)).Elem()
+var (
+	errorType   = reflect.TypeOf((*error)(nil)).Elem()
+	contextType = reflect.TypeOf((*context.Context)(nil)).Elem()
+)
 
 type Function = func(params ...interface{}) (interface{}, error)
 
 func Run(program *Program, env interface{}) (interface{}, error) {
+	return RunWithContext(context.Background(), program, env)
+}
+
+func RunWithContext(ctx context.Context, program *Program, env interface{}) (interface{}, error) {
 	if program == nil {
 		return nil, fmt.Errorf("program is nil")
 	}
-
-	vm := VM{}
+	vm := VM{ctx: ctx}
```

Comment:
Currently, vm doesn't do anything with context. Only passed to some functions. 

But **expr.RunWithContext** sounds like VM should be stopped (for example on deadline).